File: aula-extra-est-main/BackendAulaExtra/src/Modules/Student/DTOs/SubmittedProfessorEvaluationDto.cs
using System;

namespace ConfidantPostgreSQL.Modules.Student.DTOs
{
    public class SubmittedProfessorEvaluationDto
    {
        public Guid ProfessorFeedbackId { get; set; }
        public Guid ProfessorId { get; set; }
        public string ProfessorName { get; set; } = string.Empty;
        public int? Rating { get; set; }
        public string? Comments { get; set; }
        public DateTime? CreatedAt { get; set; }
    }
}

File: aula-extra-est-main/FrontendAulaExtra/AulaExtra/lib/core/data/student_professor_evaluations/dtos/submitted_professor_evaluation_dto.dart
class SubmittedProfessorEvaluationDto {
  SubmittedProfessorEvaluationDto({
    required this.professorFeedbackId,
    required this.professorId,
    required this.professorName,
    required this.rating,
    required this.comments,
    required this.createdAt,
  });

  final String professorFeedbackId;
  final String professorId;
  final String professorName;
  final int? rating;
  final String? comments;
  final DateTime? createdAt;

  factory SubmittedProfessorEvaluationDto.fromJson(Map<String, dynamic> json) {
    return SubmittedProfessorEvaluationDto(
      professorFeedbackId: json['professorFeedbackId']?.toString() ?? '',
      professorId: json['professorId']?.toString() ?? '',
      professorName: json['professorName']?.toString() ?? '',
      rating: json['rating'] is int ? json['rating'] as int : int.tryParse(json['rating']?.toString() ?? ''),
      comments: json['comments']?.toString(),
      createdAt: json['createdAt'] == null ? null : DateTime.tryParse(json['createdAt'].toString()),
    );
  }
}
